### PR TITLE
Added bot mining

### PR DESCRIPTION
**Pull request by linaori**
Bot behavior is a bit quirky at times, but it seems to work just fine. Also fixed a bug that let bots mine tiles anyway, despite being support beams.

You can see how it works here: https://www.youtube.com/watch?v=7bJtkjlgCnM

### CODE REVIEW

**Comment by Jayefuu** [comment]
Yay! I really dig this feature. Are you doing anything balance-wise to make them super slow to start with, with tech upgrades improving it a lot?

File: map_gen/Diggy/Feature/DiggyCaveCollapse.lua [hunk 1]
```diff
@@ -360,7 +350,12 @@ function DiggyCaveCollapse.register(cfg)
     Event.add(defines.events.on_surface_created, on_surface_created)
 
     Event.add(defines.events.on_marked_for_deconstruction, function (event)
-        if (nil ~= support_beam_entities[event.entity.name]) then
+        local name = event.entity.name
+        if name == 'sand-rock-big' or name == 'rock-huge' then
+            return
+        end
+
+        if name == 'deconstructible-tile-proxy' or nil ~= support_beam_entities[name] then
```
Comment (Valansch) [inline map_gen/Diggy/Feature/DiggyCaveCollapse.lua:358]: `nil ~= support_beam_entities[name]`
can just be
`not support_beam_entities[name]`

**Review by Valansch — APPROVED** [review]
I didnt understand why you are creating the rock again after removing it. So i tried it. THIS IS AWSOME

**Comment by plague006** [comment]
Tested this some: rooms generate fine, biters generate, collapses work. LGTM

**Review by mheguy — APPROVED** [review]
Tested this some: rooms generate fine, biters generate, collapses work. LGTM

File: map_gen/Diggy/Feature/DiggyCaveCollapse.lua (hunk 1 above)
Comment (linaori) [reply map_gen/Diggy/Feature/DiggyCaveCollapse.lua:358]: old code, will fix this

**Comment by linaori** [comment]
@Jayefuu later this evening I will work on the following:
 - set the base damage to 4 (10 atm)
 - increase the base damage by 1 per mining productivity level. That will make it roughly 7 by the time you can make bots I think

**Comment by mheguy** [comment]
Interesting, I like the idea of increasing the value of mining productivity research.

**Pull request merged**